### ISSUE TITLE
导入其他vits模型时出错

### ISSUE THREAD

**Issue by Ealvn** [dpm12345/nonebot_plugin_tts_gal#8]
感谢大佬写的代码
但我在导入其他vits模型时会报错，具体如下：
1. [colab原代码](https://colab.research.google.com/drive/1FeuXBF_XXjYiEI11Y9UQW7YBfUZFJZbc?usp=sharing#scrollTo=UrAcl4QFJMx-)
2. [所用pth模型](https://drive.google.com/uc?id=1gaNtg4XVx0Eo4RC3gScHc3dNBWhoVdxv)
3. jason文件
`{
  "train": {
    "log_interval": 200,
    "eval_interval": 1000,
    "seed": 1234,
    "epochs": 10000,
    "learning_rate": 2e-4,
    "betas": [0.8, 0.99],
    "eps": 1e-9,
    "batch_size": 32,
    "fp16_run": true,
    "lr_decay": 0.999875,
    "segment_size": 8192,
    "init_lr_ratio": 1,
    "warmup_epochs": 0,
    "c_mel": 45,
    "c_kl": 1.0
  },
  "data": {
    "training_files":"filelists/train_filelist.txt.cleaned",
    "validation_files":"filelists/val_filelist.txt.cleaned",
    "text_cleaners":["japanese_cleaners"],
    "max_wav_value": 32768.0,
    "sampling_rate": 22050,
    "filter_length": 1024,
    "hop_length": 256,
    "win_length": 1024,
    "n_mel_channels": 80,
    "mel_fmin": 0.0,
    "mel_fmax": null,
    "add_blank": true,
    "n_speakers": 5,
    "cleaned_text": true
  },
  "model": {
    "inter_channels": 192,
    "hidden_channels": 192,
    "filter_channels": 768,
    "n_heads": 2,
    "n_layers": 6,
    "kernel_size": 3,
    "p_dropout": 0.1,
    "resblock": "1",
    "resblock_kernel_sizes": [3,7,11],
    "resblock_dilation_sizes": [[1,3,5], [1,3,5], [1,3,5]],
    "upsample_rates": [8,8,2,2],
    "upsample_initial_channel": 512,
    "upsample_kernel_sizes": [16,16,4,4],
    "n_layers_q": 3,
    "use_spectral_norm": false,
    "gin_channels": 256
  },
    "speakers": ["Kasugano Sora", "Amatsume Akira", "Yorihime Nao", "Migiwa Kazuha", "Kasugano Haruka","Kasugano Haruka"],
    "symbols": ["_", ",", ".", "!", "?", "-", "A", "E", "I", "N", "O", "Q", "U", "a", "b", "d", "e", "f", "g", "h", "i", "j", "k", "m", "n", "o", "p", "r", "s", "t", "u", "v", "w", "y", "z", "\u0283", "\u02a7", "\u2193", "\u2191", " "]
}`

4. 报错：
![%18GF)M59(ABCR5Q_KGOFU5](https://user-images.githubusercontent.com/39033468/195623657-dbc75a72-3b94-4ba5-bd60-dafc7df97023.png)

希望大佬能够指点一下，查了查pytorch手册还是不知道怎么改（

**Comment by dpm12345** [comment]
> 感谢大佬写的代码 但我在导入其他vits模型时会报错，具体如下：
> 
> 1. [colab原代码](https://colab.research.google.com/drive/1FeuXBF_XXjYiEI11Y9UQW7YBfUZFJZbc?usp=sharing#scrollTo=UrAcl4QFJMx-)
> 2. [所用pth模型](https://drive.google.com/uc?id=1gaNtg4XVx0Eo4RC3gScHc3dNBWhoVdxv)
> 3. jason文件
>    `{   "train": { "log_interval": 200, "eval_interval": 1000, "seed": 1234, "epochs": 10000, "learning_rate": 2e-4, "betas": [0.8, 0.99], "eps": 1e-9, "batch_size": 32, "fp16_run": true, "lr_decay": 0.999875, "segment_size": 8192, "init_lr_ratio": 1, "warmup_epochs": 0, "c_mel": 45, "c_kl": 1.0   },   "data": { "training_files":"filelists/train_filelist.txt.cleaned", "validation_files":"filelists/val_filelist.txt.cleaned", "text_cleaners":["japanese_cleaners"], "max_wav_value": 32768.0, "sampling_rate": 22050, "filter_length": 1024, "hop_length": 256, "win_length": 1024, "n_mel_channels": 80, "mel_fmin": 0.0, "mel_fmax": null, "add_blank": true, "n_speakers": 5, "cleaned_text": true   },   "model": { "inter_channels": 192, "hidden_channels": 192, "filter_channels": 768, "n_heads": 2, "n_layers": 6, "kernel_size": 3, "p_dropout": 0.1, "resblock": "1", "resblock_kernel_sizes": [3,7,11], "resblock_dilation_sizes": [[1,3,5], [1,3,5], [1,3,5]], "upsample_rates": [8,8,2,2], "upsample_initial_channel": 512, "upsample_kernel_sizes": [16,16,4,4], "n_layers_q": 3, "use_spectral_norm": false, "gin_channels": 256   }, "speakers": ["Kasugano Sora", "Amatsume Akira", "Yorihime Nao", "Migiwa Kazuha", "Kasugano Haruka","Kasugano Haruka"], "symbols": ["_", ",", ".", "!", "?", "-", "A", "E", "I", "N", "O", "Q", "U", "a", "b", "d", "e", "f", "g", "h", "i", "j", "k", "m", "n", "o", "p", "r", "s", "t", "u", "v", "w", "y", "z", "\u0283", "\u02a7", "\u2193", "\u2191", " "] }`
> 4. 报错：
>    ![%18GF)M59(ABCR5Q_KGOFU5](https://user-images.githubusercontent.com/39033468/195623657-dbc75a72-3b94-4ba5-bd60-dafc7df97023.png)
> 
> 希望大佬能够指点一下，查了查pytorch手册还是不知道怎么改（

我测试了一下，貌似是模型文件与配置文件的不匹配导致的

模型文件似乎是只有五个角色，而配置文件中写了`7`

你可以把配置文件的`n_speakers`改为`"n_speakers": 5`

目前这个改动是可以正常加载模型的

**Comment by dpm12345** [comment]
> > 感谢大佬写的代码 但我在导入其他vits模型时会报错，具体如下：
> > 
> > 1. [colab原代码](https://colab.research.google.com/drive/1FeuXBF_XXjYiEI11Y9UQW7YBfUZFJZbc?usp=sharing#scrollTo=UrAcl4QFJMx-)
> > 2. [所用pth模型](https://drive.google.com/uc?id=1gaNtg4XVx0Eo4RC3gScHc3dNBWhoVdxv)
> > 3. jason文件
> >    `{   "train": { "log_interval": 200, "eval_interval": 1000, "seed": 1234, "epochs": 10000, "learning_rate": 2e-4, "betas": [0.8, 0.99], "eps": 1e-9, "batch_size": 32, "fp16_run": true, "lr_decay": 0.999875, "segment_size": 8192, "init_lr_ratio": 1, "warmup_epochs": 0, "c_mel": 45, "c_kl": 1.0   },   "data": { "training_files":"filelists/train_filelist.txt.cleaned", "validation_files":"filelists/val_filelist.txt.cleaned", "text_cleaners":["japanese_cleaners"], "max_wav_value": 32768.0, "sampling_rate": 22050, "filter_length": 1024, "hop_length": 256, "win_length": 1024, "n_mel_channels": 80, "mel_fmin": 0.0, "mel_fmax": null, "add_blank": true, "n_speakers": 5, "cleaned_text": true   },   "model": { "inter_channels": 192, "hidden_channels": 192, "filter_channels": 768, "n_heads": 2, "n_layers": 6, "kernel_size": 3, "p_dropout": 0.1, "resblock": "1", "resblock_kernel_sizes": [3,7,11], "resblock_dilation_sizes": [[1,3,5], [1,3,5], [1,3,5]], "upsample_rates": [8,8,2,2], "upsample_initial_channel": 512, "upsample_kernel_sizes": [16,16,4,4], "n_layers_q": 3, "use_spectral_norm": false, "gin_channels": 256   }, "speakers": ["Kasugano Sora", "Amatsume Akira", "Yorihime Nao", "Migiwa Kazuha", "Kasugano Haruka","Kasugano Haruka"], "symbols": ["_", ",", ".", "!", "?", "-", "A", "E", "I", "N", "O", "Q", "U", "a", "b", "d", "e", "f", "g", "h", "i", "j", "k", "m", "n", "o", "p", "r", "s", "t", "u", "v", "w", "y", "z", "\u0283", "\u02a7", "\u2193", "\u2191", " "] }`
> > 4. 报错：
> >    ![%18GF)M59(ABCR5Q_KGOFU5](https://user-images.githubusercontent.com/39033468/195623657-dbc75a72-3b94-4ba5-bd60-dafc7df97023.png)
> > 
> > 希望大佬能够指点一下，查了查pytorch手册还是不知道怎么改（
> 
> 我测试了一下，貌似是模型文件与配置文件的不匹配导致的
> 
> 模型文件似乎是只有五个角色，而配置文件中写了`7`
> 
> 你可以把配置文件的`n_speakers`改为`"n_speakers": 5`
> 
> 目前这个改动是可以正常加载模型的

抱歉，我可能搞错了

**Comment by dpm12345** [comment]
> 感谢大佬写的代码 但我在导入其他vits模型时会报错，具体如下：
> 
> 1. [colab原代码](https://colab.research.google.com/drive/1FeuXBF_XXjYiEI11Y9UQW7YBfUZFJZbc?usp=sharing#scrollTo=UrAcl4QFJMx-)
> 2. [所用pth模型](https://drive.google.com/uc?id=1gaNtg4XVx0Eo4RC3gScHc3dNBWhoVdxv)
> 3. jason文件
>    `{   "train": { "log_interval": 200, "eval_interval": 1000, "seed": 1234, "epochs": 10000, "learning_rate": 2e-4, "betas": [0.8, 0.99], "eps": 1e-9, "batch_size": 32, "fp16_run": true, "lr_decay": 0.999875, "segment_size": 8192, "init_lr_ratio": 1, "warmup_epochs": 0, "c_mel": 45, "c_kl": 1.0   },   "data": { "training_files":"filelists/train_filelist.txt.cleaned", "validation_files":"filelists/val_filelist.txt.cleaned", "text_cleaners":["japanese_cleaners"], "max_wav_value": 32768.0, "sampling_rate": 22050, "filter_length": 1024, "hop_length": 256, "win_length": 1024, "n_mel_channels": 80, "mel_fmin": 0.0, "mel_fmax": null, "add_blank": true, "n_speakers": 5, "cleaned_text": true   },   "model": { "inter_channels": 192, "hidden_channels": 192, "filter_channels": 768, "n_heads": 2, "n_layers": 6, "kernel_size": 3, "p_dropout": 0.1, "resblock": "1", "resblock_kernel_sizes": [3,7,11], "resblock_dilation_sizes": [[1,3,5], [1,3,5], [1,3,5]], "upsample_rates": [8,8,2,2], "upsample_initial_channel": 512, "upsample_kernel_sizes": [16,16,4,4], "n_layers_q": 3, "use_spectral_norm": false, "gin_channels": 256   }, "speakers": ["Kasugano Sora", "Amatsume Akira", "Yorihime Nao", "Migiwa Kazuha", "Kasugano Haruka","Kasugano Haruka"], "symbols": ["_", ",", ".", "!", "?", "-", "A", "E", "I", "N", "O", "Q", "U", "a", "b", "d", "e", "f", "g", "h", "i", "j", "k", "m", "n", "o", "p", "r", "s", "t", "u", "v", "w", "y", "z", "\u0283", "\u02a7", "\u2193", "\u2191", " "] }`
> 4. 报错：
>    ![%18GF)M59(ABCR5Q_KGOFU5](https://user-images.githubusercontent.com/39033468/195623657-dbc75a72-3b94-4ba5-bd60-dafc7df97023.png)
> 
> 希望大佬能够指点一下，查了查pytorch手册还是不知道怎么改（

似乎是symbols的问题，你给我的配置信息中
`"symbols": ["_", ",", ".", "!", "?", "-", "A","E","I","N","O","Q","U","a","b","d","e","f","g","h","i","j","k","m","n","o","p","r","s","t","u","v","w","y","z","\u0283","\u02a7","\u2193","\u2191"," "]`
总共40个字符

而在原github的配置文件中
`"symbols": ["_", ",", ".", "!", "?", "-", "~","A", "E", "I", "N", "O", "Q", "U", "a", "b", "d", "e", "f", "g", "h", "i", "j", "k", "m", "n", "o", "p", "r", "s", "t", "u", "v", "w", "y", "z", "\u0283", "\u02a7", "\u2193", "\u2191", " "]`
总共41个字符

后面比前面多了一个`~`

你可以改一改试试

**Comment by Ealvn** [comment]
> > 感谢大佬写的代码 但我在导入其他vits模型时会报错，具体如下：
> > 
> > 1. [colab原代码](https://colab.research.google.com/drive/1FeuXBF_XXjYiEI11Y9UQW7YBfUZFJZbc?usp=sharing#scrollTo=UrAcl4QFJMx-)
> > 2. [所用pth模型](https://drive.google.com/uc?id=1gaNtg4XVx0Eo4RC3gScHc3dNBWhoVdxv)
> > 3. jason文件
> >    `{   "train": { "log_interval": 200, "eval_interval": 1000, "seed": 1234, "epochs": 10000, "learning_rate": 2e-4, "betas": [0.8, 0.99], "eps": 1e-9, "batch_size": 32, "fp16_run": true, "lr_decay": 0.999875, "segment_size": 8192, "init_lr_ratio": 1, "warmup_epochs": 0, "c_mel": 45, "c_kl": 1.0   },   "data": { "training_files":"filelists/train_filelist.txt.cleaned", "validation_files":"filelists/val_filelist.txt.cleaned", "text_cleaners":["japanese_cleaners"], "max_wav_value": 32768.0, "sampling_rate": 22050, "filter_length": 1024, "hop_length": 256, "win_length": 1024, "n_mel_channels": 80, "mel_fmin": 0.0, "mel_fmax": null, "add_blank": true, "n_speakers": 5, "cleaned_text": true   },   "model": { "inter_channels": 192, "hidden_channels": 192, "filter_channels": 768, "n_heads": 2, "n_layers": 6, "kernel_size": 3, "p_dropout": 0.1, "resblock": "1", "resblock_kernel_sizes": [3,7,11], "resblock_dilation_sizes": [[1,3,5], [1,3,5], [1,3,5]], "upsample_rates": [8,8,2,2], "upsample_initial_channel": 512, "upsample_kernel_sizes": [16,16,4,4], "n_layers_q": 3, "use_spectral_norm": false, "gin_channels": 256   }, "speakers": ["Kasugano Sora", "Amatsume Akira", "Yorihime Nao", "Migiwa Kazuha", "Kasugano Haruka","Kasugano Haruka"], "symbols": ["_", ",", ".", "!", "?", "-", "A", "E", "I", "N", "O", "Q", "U", "a", "b", "d", "e", "f", "g", "h", "i", "j", "k", "m", "n", "o", "p", "r", "s", "t", "u", "v", "w", "y", "z", "\u0283", "\u02a7", "\u2193", "\u2191", " "] }`
> > 4. 报错：
> >    ![%18GF)M59(ABCR5Q_KGOFU5](https://user-images.githubusercontent.com/39033468/195623657-dbc75a72-3b94-4ba5-bd60-dafc7df97023.png)
> > 
> > 希望大佬能够指点一下，查了查pytorch手册还是不知道怎么改（
> 
> 似乎是symbols的问题，你给我的配置信息中 `"symbols": ["_", ",", ".", "!", "?", "-", "A","E","I","N","O","Q","U","a","b","d","e","f","g","h","i","j","k","m","n","o","p","r","s","t","u","v","w","y","z","\u0283","\u02a7","\u2193","\u2191"," "]` 总共40个字符
> 
> 而在原github的配置文件中 `"symbols": ["_", ",", ".", "!", "?", "-", "~","A", "E", "I", "N", "O", "Q", "U", "a", "b", "d", "e", "f", "g", "h", "i", "j", "k", "m", "n", "o", "p", "r", "s", "t", "u", "v", "w", "y", "z", "\u0283", "\u02a7", "\u2193", "\u2191", " "]` 总共41个字符
> 
> 后面比前面多了一个`~`
> 
> 你可以改一改试试

感谢大佬，明白了
这部分我是复制其他模型的，没想到不一定通用
